contrib/OtlpHttp/Exporter.php

```diff
@@ -0,0 +1,210 @@
+     */
+    public function __construct(
+        ClientInterface $client,
+        RequestFactoryInterface $requestFactory,
+        StreamFactoryInterface $streamFactory,
+        SpanConverter $spanConverter = null
+    ) {
+
+        // Set default values based on presence of env variable
+        $this->endpointUrl = getenv('OTEL_EXPORTER_OTLP_ENDPOINT') ?: 'https://localhost:55681/v1/traces';
+        $this->protocol = getenv('OTEL_EXPORTER_OTLP_PROTOCOL') ?: 'http/protobuf';
+        $this->certificateFile = getenv('OTEL_EXPORTER_OTLP_CERTIFICATE') ?: 'none';
+        $this->headers = getenv('OTEL_EXPORTER_OTLP_HEADERS') ?: 'none';
+        $this->compression = getenv('OTEL_EXPORTER_OTLP_COMPRESSION') ?: 'none';
+        $this->timeout =(int) getenv('OTEL_EXPORTER_OTLP_TIMEOUT') ?: 10;
+
+        $this->client = $client;
+        $this->requestFactory = $requestFactory;
+        $this->streamFactory = $streamFactory;
+        $this->spanConverter =  $spanConverter ?? new SpanConverter();
+
+        if ($this->protocol != 'http/protobuf') {
+            throw new InvalidArgumentException('Invalid OTLP Protocol Specified');
+        }
+
+        $parsedDsn = parse_url($this->endpointUrl);
+
+        if (!is_array($parsedDsn)) {
+            throw new InvalidArgumentException('Unable to parse provided DSN');
+        }
+
+        if (
+            !isset($parsedDsn['scheme'])
+            || !isset($parsedDsn['host'])
+            || !isset($parsedDsn['port'])
+            || !isset($parsedDsn['path'])
+        ) {
+            throw new InvalidArgumentException('Endpoint should have scheme, host, port and path');
+        }
+    }
+
+    /**
+     * Exports the provided Span data via the OTLP protocol
+     *
+     * @param iterable<API\Span> $spans Array of Spans
+     * @return int return code, defined on the Exporter interface
+     */
+    public function export(iterable $spans): int
+    {
+        if (!$this->running) {
+            return Exporter::FAILED_NOT_RETRYABLE;
+        }
+
+        if (empty($spans)) {
+            return Trace\Exporter::SUCCESS;
+        }
+
+        $resourcespans = [$this->spanConverter->as_otlp_resource_span($spans)];
+
+        $exportrequest = new ExportTraceServiceRequest([
+            'resource_spans' => $resourcespans,
+        ]);
+
+        $bytes = $exportrequest->serializeToString();
+
+        try {
+            $request = $this->requestFactory
+                ->createRequest('POST', $this->endpointUrl)
+                ->withHeader('content-type', 'application/x-protobuf');
+
+            foreach ($this->processHeaders($this->headers) as $header => $value) {
+                $request = $request->withHeader($header, $value);
+            }
+
+            if ($this->compression === 'gzip') {
+                // TODO: Add Tests
+                $body = $this->streamFactory->createStream(gzencode($bytes));
+                $request = $request->withHeader('Content-Encoding', 'gzip');
```

Comment:
I wasn't sure if this was the correct way to gzip the request body.

Reply:
Seems reasonable to me (the `gzencode` library is used often).  You could always echo this out for your test and ensure that we have an encoded sample.